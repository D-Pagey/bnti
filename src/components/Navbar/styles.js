import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { margin, padding } from 'polished';
import { border, colour, fontSize, fontWeight, spacing } from '../../tokens';

export const List = styled.ul`
    align-items: center;
    display: flex;
    list-style: none;
    ${margin(0)};
    ${padding(0)};
`;

export const ListItem = styled.li`
    ${margin(0, spacing.big, 0, 0)}

    &:hover {
        color: ${colour.white};
    }
`;

export const NavbarLink = styled(NavLink)`
    color: inherit;
    font-size: ${fontSize.medium};
    font-weight: ${fontWeight.medium};
    text-decoration: none;
`;

export const NavbarButton = styled.button.attrs({
    type: 'button'
})`
    background-color: ${colour.black};
    border: 0;
    border-radius: ${border.radiusLarge};
    color: ${colour.white};
    cursor: pointer;
    font-size: ${fontSize.medium};
    font-weight: ${fontWeight.hair};
    ${padding(spacing.tiny, spacing.small)};
`;
