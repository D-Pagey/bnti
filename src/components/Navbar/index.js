import React, { useContext } from 'react';
import AuthContext from '../../contexts/auth';
import * as S from './styles';

const Navbar = () => {
    const { authSession, authLogout } = useContext(AuthContext);

    return (
        <S.List>
            <S.ListItem>
                <S.NavbarLink to="/about">About</S.NavbarLink>
            </S.ListItem>

            <S.ListItem>
                <S.NavbarLink to="/tournaments">Tournaments</S.NavbarLink>
            </S.ListItem>

            <S.ListItem>
                <S.NavbarLink to="/stats">Stats</S.NavbarLink>
            </S.ListItem>

            <S.ListItem>
                <S.NavbarLink to="/signup">Sign Up</S.NavbarLink>
            </S.ListItem>

            <S.ListItem>
                {authSession ? (
                    <S.NavbarButton onClick={authLogout} data-testid="logoutButton">
                        Logout
                    </S.NavbarButton>
                ) : (
                    <S.NavbarLink to="login">
                        <S.NavbarButton>Login</S.NavbarButton>
                    </S.NavbarLink>
                )}
            </S.ListItem>
        </S.List>
    );
};

export default Navbar;
