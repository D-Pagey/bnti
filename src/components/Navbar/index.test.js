import React from 'react';
import AuthContext from '../../contexts/auth';
import Navbar from '.';

describe('Navbar component', () => {
    it('should render', () => {
        const { container } = renderWithRouter(<Navbar />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with logout button', () => {
        const { getByTestId } = renderWithRouter(
            <AuthContext.Provider value={{ authSession: {} }}>
                <Navbar />
            </AuthContext.Provider>
        );

        getByTestId('logoutButton');
    });
});
