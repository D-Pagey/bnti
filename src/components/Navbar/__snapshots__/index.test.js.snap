// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c1 {
  margin-right: 2rem;
}

.c1:hover {
  color: white;
}

.c2 {
  color: inherit;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  background-color: black;
  border: 0;
  border-radius: 16px;
  color: white;
  cursor: pointer;
  font-size: 1.2rem;
  font-weight: 100;
  padding: 5px 15px;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <a
      class="c2"
      href="/about"
    >
      About
    </a>
  </li>
  <li
    class="c1"
  >
    <a
      class="c2"
      href="/tournaments"
    >
      Tournaments
    </a>
  </li>
  <li
    class="c1"
  >
    <a
      class="c2"
      href="/stats"
    >
      Stats
    </a>
  </li>
  <li
    class="c1"
  >
    <a
      class="c2"
      href="/login"
    >
      <button
        class="c3"
        type="button"
      >
        Login
      </button>
    </a>
  </li>
</ul>
`;
